### DISCUSSION TITLE
Setting username and password for SPARQL repository connections

### DISCUSSION THREAD

**Discussion by reckart** [eclipse-rdf4j/rdf4j#3325] [Q&A]
It looks like the SPARQLRepository supports setting a username/password for accessing a remote endpoint, but the SPARQLRepositoryConfig class seems not to support it. Is there a way to set a username/password for SPARQLRepositories which are configured through a repository manager?

Related:
* https://github.com/inception-project/inception/issues/2589
* https://github.com/eclipse/rdf4j/issues/1220

**Comment by abrokenjester** [comment]
It's possible to still set the username/password programmatically after you've retrieved the SPARQLRepository instance from the manager, via `#setUsernameAndPassword`, but there is currently no option to define this in a configuration file I'm afraid. It shouldn't be hard to add, the HTTPRepositoryConfig already has this built in, it's pretty much the same mechanism copied over. We'd welcome a PR if you can.
